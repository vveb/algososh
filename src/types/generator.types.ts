import { ViewItem } from "./view.types";

export type IterableViewWithNumbers = Generator<ViewItem<number>[], void, never>;
export type IterableViewWithStrings = Generator<ViewItem<string>[], void, never>;