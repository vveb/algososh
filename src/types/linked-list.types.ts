export interface ILinkedListNode<T> {
  value: T,
  next: ILinkedListNode<T> | null,
};

export interface ILinkedList<T> {
  prepend: (item: T) => void,
  append: (item: T) => void,
  addByIndex: (item: T, index: number) => void,
  deleteByIndex: (index: number) => void,
  deleteHead: () => void,
  deleteTail: () => void,
  toArray: () => ILinkedListNode<T>[] | null,
  isEmpty: boolean,
};

export enum LinkedListActions {
  AddToHead = "addToHead",
  AddToTail = "addToTail",
  DeleteFromHead = "deleteFromHead",
  DeleteFromTail = "deleteFromTail",
  AddAtIndex = 'addAtIndex',
  DeleteFromIndex = 'deleteFromIndex',
};

export type LinkedListIsAnimated = {
  isAddToHeadAnimating: boolean,
  isAddToTailAnimating: boolean,
  isDeleteFromHeadAnimating: boolean,
  isDeleteFromTailAnimating: boolean,
  isAddAtIndexAnimating: boolean,
  isDeleteFromIndexAnimating: boolean,
};