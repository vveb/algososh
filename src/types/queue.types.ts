export interface IQueueNode<T> {
  value: T,
  next: IQueueNode<T> | null,
};

export interface IQueue<T> {
  enqueue: (item: T) => void;
  dequeue: () => T | null;
  clear: () => void;
  isEmpty: boolean,
};