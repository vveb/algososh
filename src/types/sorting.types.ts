export enum SortingType {
  Selection = "selection",
  Bubble = "bubble",
}