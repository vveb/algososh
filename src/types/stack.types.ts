export interface IStack<T> {
  push: (item: T) => void;
  pop: () => T | undefined;
  clear: () => void;
  getElements: () => T[] | null;
  getSize: () => number;
};

// Для связного списка...
export interface RegularStackNode<T> {
  value: T,
  on: RegularStackNode<T>,
};

export interface IStack2<T, N extends RegularStackNode<T> = RegularStackNode<T>> {
  push: (item: T) => void;
  pop: () => T | undefined;
  clear: () => void;
  isEmpty: boolean;
};