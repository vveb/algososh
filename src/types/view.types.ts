import { ElementStates } from "./element-states";

export type ViewItem<T extends string | number> = {
  value: T,
  state: ElementStates,
};