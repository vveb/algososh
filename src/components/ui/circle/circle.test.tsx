import renderer from 'react-test-renderer';
import { Circle } from "./circle";
import { ElementStates } from '../../../types/element-states';

describe('Circle rendering', () => {
  it('Circle without letter rendering successfull', () => {
    const tree = renderer
    .create(<Circle />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle with letter rendering successfull', () => {
    const tree = renderer
    .create(<Circle letter='A'/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle with head-string rendering successfull', () => {
    const tree = renderer
    .create(<Circle head='head'/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle with head-reactElement rendering successfull', () => {
    const tree = renderer
    .create(<Circle head={<Circle />}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle with tail-string rendering successfull', () => {
    const tree = renderer
    .create(<Circle tail='head'/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle with tail-reactElement rendering successfull', () => {
    const tree = renderer
    .create(<Circle tail={<Circle />}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle with index rendering successfull', () => {
    const tree = renderer
    .create(<Circle index={0}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle with isSmall prop === true rendering successfull', () => {
    const tree = renderer
    .create(<Circle isSmall={true}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle with default state rendering successfull', () => {
    const tree = renderer
    .create(<Circle state={ElementStates.Default}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle with changing state rendering successfull', () => {
    const tree = renderer
    .create(<Circle state={ElementStates.Changing}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle with modified state rendering successfull', () => {
    const tree = renderer
    .create(<Circle state={ElementStates.Modified}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});