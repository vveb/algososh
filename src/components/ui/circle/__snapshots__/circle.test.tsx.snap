// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle rendering Circle with changing state rendering successfull 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering Circle with default state rendering successfull 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering Circle with head-reactElement rendering successfull 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  >
    <div
      className="undefined "
    >
      <div
        className="text text_type_input text_color_input mb-4 undefined undefined undefined"
      />
      <div
        className="undefined   undefined"
      >
        <p
          className="text text_type_circle text_color_input undefined"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 undefined undefined"
      />
      <div
        className="text text_type_input text_color_input mt-4 undefined undefined undefined"
      />
    </div>
  </div>
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering Circle with head-string rendering successfull 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  >
    head
  </div>
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering Circle with index rendering successfull 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering Circle with isSmall prop === true rendering successfull 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined  undefined undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering Circle with letter rendering successfull 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering Circle with modified state rendering successfull 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering Circle with tail-reactElement rendering successfull 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  >
    <div
      className="undefined "
    >
      <div
        className="text text_type_input text_color_input mb-4 undefined undefined undefined"
      />
      <div
        className="undefined   undefined"
      >
        <p
          className="text text_type_circle text_color_input undefined"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 undefined undefined"
      />
      <div
        className="text text_type_input text_color_input mt-4 undefined undefined undefined"
      />
    </div>
  </div>
</div>
`;

exports[`Circle rendering Circle with tail-string rendering successfull 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  >
    head
  </div>
</div>
`;

exports[`Circle rendering Circle without letter rendering successfull 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering should render Circle component with changing state 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering should render Circle component with default state 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering should render Circle component with head-reactElement 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  >
    <div
      className="undefined "
    >
      <div
        className="text text_type_input text_color_input mb-4 undefined undefined undefined"
      />
      <div
        className="undefined   undefined"
      >
        <p
          className="text text_type_circle text_color_input undefined"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 undefined undefined"
      />
      <div
        className="text text_type_input text_color_input mt-4 undefined undefined undefined"
      />
    </div>
  </div>
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering should render Circle component with head-string 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  >
    head
  </div>
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering should render Circle component with index 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering should render Circle component with isSmall prop === true 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined  undefined undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering should render Circle component with letter 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering should render Circle component with modified state 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle rendering should render Circle component with tail-reactElement 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  >
    <div
      className="undefined "
    >
      <div
        className="text text_type_input text_color_input mb-4 undefined undefined undefined"
      />
      <div
        className="undefined   undefined"
      >
        <p
          className="text text_type_circle text_color_input undefined"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 undefined undefined"
      />
      <div
        className="text text_type_input text_color_input mt-4 undefined undefined undefined"
      />
    </div>
  </div>
</div>
`;

exports[`Circle rendering should render Circle component with tail-string 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  >
    head
  </div>
</div>
`;

exports[`Circle rendering should render Circle component without letter 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;
